### PR TITLE
Login and logout

### LINKED ISSUE #3: [employee] Login and logout

### PR DESCRIPTION
**Pull request by magley**
The class `GenericEmployeeView` is a placeholder to test logging out.

Having a different `JFrame` for the login form and the actual application is fine (the alternative, using different jpanels, is tedious, because of how poorly designed Swing is).

Closes #3

### CODE REVIEW

File: src/app/Main.java [hunk 1]
```diff
@@ -22,26 +27,21 @@ public static void main(String[] args) {
...
+		EmployeeController employeeController = new EmployeeController(employeeService);
 		
-		for (int i = 0; i < 4; i++) {
-			Employee e = new Employee("person" + i, "lastname" + i, accountService.get(i), Employee.Role.values()[1 + i]);
-			employeeService.add(e);
-		}
+		JFrame frame = new EmployeeLoginView(accountController, employeeController);
+		frame.setSize(800, 600);  
+        frame.setLocationRelativeTo(null);  
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
```
Comment (swopl) [inline src/app/Main.java:36]: Some of these indents are 4 spaces and some are tabs. Please convert all to one type. 4 spaces?

File: src/app/Main.java [hunk 2]
```diff
@@ -22,26 +27,21 @@ public static void main(String[] args) {
...
-		}
+		JFrame frame = new EmployeeLoginView(accountController, employeeController);
+		frame.setSize(800, 600);  
+        frame.setLocationRelativeTo(null);  
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);  
+        frame.setVisible(true);
+        
+        frame.addWindowListener(new java.awt.event.WindowAdapter() {
+            @Override
+            public void windowClosing(java.awt.event.WindowEvent windowEvent) {
```
Comment (swopl) [inline src/app/Main.java:41]: Import these instead of writing full name

File: src/core/account/AccountController.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package core.account;
+
+import core.account.exception.AccountNotFoundException;
```
Comment (swopl) [inline src/core/account/AccountController.java:3]: Should controllers be in core package or with view?

File: src/desktop/employee/EmployeeLoginView.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+		add(btnLogin, "cell 1 4 2 1,growx");
+	}
+	
+	private void login() {
+		String email = txtEmail.getText();
+		String password = txtPassword.getText();
+		
+		try {
+			Account acc = accountController.login(email, password);
+			Employee e = employeeController.getByAccount(acc);
```
Comment (swopl) [inline src/desktop/employee/EmployeeLoginView.java:71]: Maybe flip the direction Employee -> Acc so it is Acc -> Employee? Idk...

File: src/desktop/employee/EmployeeLoginView.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+		add(txtPassword, "cell 2 2,growx");
+		txtPassword.setColumns(10);
+		
+		btnLogin = new JButton("Login");
+		btnLogin.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				login();
+			}
+		});
+		add(btnLogin, "cell 1 4 2 1,growx");
```
Comment (swopl) [inline src/desktop/employee/EmployeeLoginView.java:62]: I don't remember how to do this in swing, but can you make it so on enter pressed when in the TextFields it tries a login? Also allow users to move to login using TAB key and press enter there? I don't remember how to do this in swing

File: src/desktop/employee/EmployeeLoginView.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+		add(txtEmail, "cell 2 1,growx");
+		txtEmail.setColumns(10);
+		
+		JLabel lblPassword = new JLabel("Password:");
+		lblPassword.setHorizontalAlignment(SwingConstants.CENTER);
+		add(lblPassword, "cell 1 2,growx");
+		
+		txtPassword = new JTextField();
+		add(txtPassword, "cell 2 2,growx");
+		txtPassword.setColumns(10);
```
Comment (swopl) [inline src/desktop/employee/EmployeeLoginView.java:54]: Make text and input fields bigger

Comment (swopl) [inline src/desktop/employee/EmployeeLoginView.java:54]: Use passwordfield for password. Or however it is called

**Review by swopl — CHANGES_REQUESTED** [review]

File: src/app/Main.java (hunk 1 above)
Comment (magley) [reply src/app/Main.java]: Fixed in [b58d96a](https://github.com/magley/sims-tollbooth/pull/17/commits/b58d96aa0a3088043382aa73796747a1a20add16)

File: src/app/Main.java (hunk 2 above)
Comment (magley) [reply src/app/Main.java]: Fixed in [ec55ddf](https://github.com/magley/sims-tollbooth/pull/17/commits/ec55ddf3ff25030ef3fb01b7dd60cc0344c8ea5d)

File: src/desktop/employee/EmployeeLoginView.java (hunk 5 above)
Comment (magley) [reply src/desktop/employee/EmployeeLoginView.java]: By default, you can navigate between GUI elements with Tab. You can simulate a button push with Space. 

I added the ability to try to login on enter, regardless of which GUI element is focused, in [7fb3ac3](https://github.com/magley/sims-tollbooth/pull/17/commits/7fb3ac329f55a8226eea30088df9e8097f241115).

File: src/desktop/employee/EmployeeLoginView.java (hunk 6 above)
Comment (magley) [reply src/desktop/employee/EmployeeLoginView.java]: They grow with the window, this should be big enough already?

image

Comment (magley) [reply src/desktop/employee/EmployeeLoginView.java]: Done in [88c30a1](https://github.com/magley/sims-tollbooth/pull/17/commits/88c30a1a11622129f3785161a3dbf0b5b427ea1a)

File: src/desktop/employee/EmployeeLoginView.java (hunk 4 above)
Comment (magley) [reply src/desktop/employee/EmployeeLoginView.java]: Done in [af89296](https://github.com/magley/sims-tollbooth/pull/17/commits/af892964a2135e44e1996979d89f0989f18460ab)

File: src/core/account/AccountController.java (hunk 3 above)
Comment (swopl) [reply src/core/account/AccountController.java:3]: It's ok like this

File: src/desktop/employee/EmployeeLoginView.java (hunk 6 above)
Comment (magley) [reply src/desktop/employee/EmployeeLoginView.java]: I made the empty space on the left and right side not grow anymore, but preferably take up 20% of window space in [e16a612](https://github.com/magley/sims-tollbooth/pull/17/commits/e16a6129c73b91a3e8727f89a3b78b1a01f69ece).

Comment (magley) [reply src/desktop/employee/EmployeeLoginView.java]: I don't know. I've never done something like that in Swing. Either way, I don't think it's very important for this PR. We can open up another issue just for making the GUI accessible.

Comment (swopl) [reply src/desktop/employee/EmployeeLoginView.java]: Ok, open the issue and you can merge this now then

**Pull request merged**